imap_processing/hi/utils.py

```diff
@@ -67,6 +68,78 @@ class HiConstants:
     TOF3_BAD_VALUES = (1023,)
 
 
+def parse_filename_like(filename_like: str) -> re.Match:
+    """
+    Parse a filename like string.
+
+    This function is based off of the more strict regex parsing of IMAP science
+    product filenames found in the `imap_data_access` package `ScienceFilePath`
+    class. This function implements a more relaxed regex that can be used on
+    `Logical_source` or `Logical_file_id` found in the CDF file. The required
+    components in the input string are `mission`, `instrument`, `data_level`,
+    and `descriptor`.
+
+    Parameters
+    ----------
+    filename_like : str
+        A filename like string. This includes `Logical_source` or `Logical_file_id`
+        strings.
+
+    Returns
+    -------
+    match : re.Match
+        A dictionary like re.Match object resulting from parsing the input string.
+
+    Raises
+    ------
+    ValueError if the regex fails to match the input string.
+    """
+    regex_str = (
+        r"^(?P<mission>imap)_"  # Required mission
+        r"(?P<instrument>[^_]+)_"  # Required instrument
+        r"(?P<data_level>[^_]+)_"  # Required data level
+        r"((?P<sensor>\d{2}sensor)?-)?"  # Optional sensor number
+        r"(?P<descriptor>[^_]+)"  # Required descriptor
+        r"(_(?P<start_date>\d{8}))?"  # Optional start date
+        r"(-repoint(?P<repointing>\d{5}))?"  # Optional repointing field
+        r"(?:_v(?P<version>\d{3}))?"  # Optional version
+        r"(?:\.(?P<extension>cdf|pkts))?$"  # Optional extension
+    )
```

Comment:
Do you need other fields or can you do something similar to [this](https://github.com/IMAP-Science-Operations-Center/imap_processing/blob/94324f5095a5f3d6994cf72b4d951ee1ca18eb20/imap_processing/cdf/utils.py#L97) and then apply regex on `descriptor`? I am afraid of losing track of this in the future. I am wondering now if it's an feature we should implement in `imap-data-access`, especially update [this function](https://github.com/IMAP-Science-Operations-Center/imap-data-access/blob/main/imap_data_access/file_validation.py#L265).
```suggestion
    regex_str = (
        r"((?P<sensor>\d{2}sensor)?-)?"  # Optional sensor number
        r"(?P<descriptor>[^_]+)"  # Required descriptor
    )
```

Reply:
Yeah, perhaps it is overkill to parse all of the fields. I doubt I will need the `mission` or `instrument` field but I could see needing the `start_date` or `repoint`. I based this regex on the one in the `ScienceFilePath` class. I don't think that one should be modified because it needs to be very strict to have invalid file names fail where I don't want to be strict. Parsing using `split()` could be done but it is lots of hard coding indices and gets messy with some of the optional fields.

Reply:
If you like to keep it as it is, that's ok with me. I do want @laspsandoval and @sdhoyt to review these two function especially and see if they can benefit from it. If so, then we may want to move this in common utils function.

Reply:
I haven't come across a situation where I need to parse anything from the logical source. I agree with Tim though about the `start_date` and `pointing`. I could see parsing those two fields being useful for instruments that need to reprocess because there's data spread across CCSDS files for two pointings and they need to put some bits of data from Day 2 into the Day 1 data product. In that case I think there would be multiple L0 dependency files with different dates.

Reply:
@tech3371, I decided to move this function into the cdf/utils.py module.

Reply:
Tim, that's a good place to put it since it's related to CDF attrs. Cool. I will approve this PR then. Then if others find that function helpful, it's there to use.